tests/src/JIT/HardwareIntrinsics/X86/Sse2/AddSaturate.cs

```diff
@@ -0,0 +1,119 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using System;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+namespace IntelHardwareIntrinsicTest
+{
+    internal static partial class Program
+    {
+        const int Pass = 100;
+        const int Fail = 0;
+
+        static unsafe int Main(string[] args)
+        {
+            int testResult = Pass;
+            int testsCount = 21;
+            string methodUnderTestName = nameof(Sse2.AddSaturate);
+
+            if (Sse2.IsSupported)
+            {
+                using (var shortTable = TestTableSse2<short>.Create(testsCount))
+                using (var ushortTable = TestTableSse2<ushort>.Create(testsCount))
+                using (var sbyteTable = TestTableSse2<sbyte>.Create(testsCount))
+                using (var byteTable = TestTableSse2<byte>.Create(testsCount))
+                {
+                    for (int i = 0; i < testsCount; i++)
+                    {
+                        (Vector128<short>, Vector128<short>, Vector128<short>) value = shortTable[i];
+                        var result = Sse2.AddSaturate(value.Item1, value.Item2);
+                        shortTable.SetOutArray(result);
+                    }
+
+                    for (int i = 0; i < testsCount; i++)
+                    {
+                        (Vector128<ushort>, Vector128<ushort>, Vector128<ushort>) value = ushortTable[i];
+                        var result = Sse2.AddSaturate(value.Item1, value.Item2);
+                        ushortTable.SetOutArray(result);
+                    }
+
+                    for (int i = 0; i < testsCount; i++)
+                    {
+                        (Vector128<sbyte>, Vector128<sbyte>, Vector128<sbyte>) value = sbyteTable[i];
+                        var result = Sse2.AddSaturate(value.Item1, value.Item2);
+                        sbyteTable.SetOutArray(result);
+                    }
+
+                    for (int i = 0; i < testsCount; i++)
+                    {
+                        (Vector128<byte>, Vector128<byte>, Vector128<byte>) value = byteTable[i];
+                        var result = Sse2.AddSaturate(value.Item1, value.Item2);
+                        byteTable.SetOutArray(result);
+                    }
+
+                    CheckMethod<short> checkInt16 = (short x, short y, short z, ref short a) =>
+                    {
+                        int value = x + y;
+                        a = (short) (value >= short.MaxValue ? short.MaxValue : value <= short.MinValue ? short.MinValue : value);
```

Comment:
Why not Math.Min and Math.Max?

Reply:
I have not intended to write concise operations but rather `explicit` operations demonstrating functionality - users tend to look into tests for clues about usage.

Reply:
Yes. I was just indicating that the following might be clearer/easier to read:
```C#
int value = x + y;
    value = Math.Max(value, short.MinValue);
    value = Math.Min(value, short.MaxValue);
a = (short)(value);
```

Reply:
Agree, this is much easier to read.